File: benchmarks/bench_5.go
package main

import (
	"fmt"
	"time"
)

func boolean_logic() int64 {
    i := int64(0)
    c := int64(0)
    for i < 2000000 {
        b1 := (i / 2) == 0
        b2 := (i / 3) == 0
        if b1 && !b2 {
            c += 1
        }
        i += 1
    }
    return c
}

func main() {
    t1 := time.Now()
    res := boolean_logic()
    t2 := time.Now()
    fmt.Println(res)
    fmt.Printf("%.6f\n", t2.Sub(t1).Seconds())
}

File: benchmarks/bench_5.py
import time
import sys
sys.setrecursionlimit(200000)

def boolean_logic():
    i = 0
    c = 0
    while i < 2000000:
        b1 = (i // 2) == 0
        b2 = (i // 3) == 0
        if b1 and not b2:
            c += 1
        i += 1
    return c


t1 = time.perf_counter()
res = boolean_logic()
t2 = time.perf_counter()
print(res)
print(f"{t2 - t1:.6f}")
